File: primos.py
import sys
import threading

def is_prime(num):
    if num < 2:
        return False
    for i in range(2, int(num**0.5) + 1):
        if num % i == 0:
            return False
    return True

def print_primes(n):
    primes = [str(num) for num in range(1, n+1) if is_prime(num)]
    print(' '.join(primes))

if __name__ == '__main__':
    if len(sys.argv) != 2:
        print('Uso: python primos.py <numero>')
        sys.exit(1)

    try:
        num = int(sys.argv[1])
        if num < 1:
            print('O número deve ser maior ou igual a 1.')
            sys.exit(1)
    except ValueError:
        print('O número fornecido não é válido.')
        sys.exit(1)

    t = threading.Thread(target=print_primes, args=(num,))
    t.start()
    t.join()

File: primos.c
#include <stdio.h>
#include <stdlib.h>
#include <pthread.h>

void* print_primes(void* arg) {
    int n = *((int*)arg);
    for (int i = 1; i <= n; i++) {
        int is_prime = 1;
        for (int j = 2; j < i; j++) {
            if (i % j == 0) {
                is_prime = 0;
                break;
            }
        }
        if (is_prime) {
            printf("%d ", i);
        }
    }
    printf("\n");
    pthread_exit(NULL);
}

int main(int argc, char* argv[]) {
    if (argc != 2) {
        printf("Uso: ./primos <numero>\n");
        return 1;
    }

    int n = atoi(argv[1]);
    if (n <= 0) {
        printf("O número deve ser maior que zero.\n");
        return 1;
    }

    pthread_t thread;
    pthread_create(&thread, NULL, print_primes, (void*)&n);
    pthread_join(thread, NULL);

    return 0;
}